File: 2024/March/3-11 791 Custom Sort String/StringSort.js
/**
 * @param {string} order
 * @param {string} s
 * @return {string}
 */
var customSortString = function(order, s) {
    const map = new Map();
    var priority = 1

    Array.from(order).forEach((c) => { map.set(c, priority++); });
    
    return Array.from(s).sort((a, b) => {
        return (map.get(a) || 0) - (map.get(b) || 0);
    }).join('');
};

File: 2024/March/3-11 791 Custom Sort String/StringSort.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        priorityMap = {}
        priority = 1
        for i in range(len(order)):
            priorityMap[order[i]] = priority
            priority += 1

        result = list(s)
        result.sort(key = lambda c : priorityMap.get(c, 0))

        return ''.join(result)